test/run_test.py

```diff
@@ -1054,6 +1154,14 @@ def run_test_module(test: str, test_directory: str, options) -> Optional[str]:
     return message
 
 
+def mp_run_test_module(test, test_directory, options):
+    log_fd, log_path = tempfile.mkstemp(dir=REPO_ROOT / "test" / "test-reports",
```

Comment:
Same comment as above, I have a feeling that temp file should be handled internally inside the handler, so there is no need to pass the temp file around and just clean up test-reports directory at the end